.github/actions/setup-nx/action.yml

```diff
@@ -155,7 +155,15 @@ runs:
           id: yarn_install_slow
           shell: bash
           run: |
-              (yarn check --integrity && yarn postinstall) || yarn install --ci
+              if (yarn check --integrity) ; then
+                yarn postinstall
+              else
+                yarn install --ci
+
+                # Ensure that the integrity check passes after install to avoid
+                # cache polution with entries that will fail future integrity checks.
+                yarn check --integrity
```

Comment:
I discovered that if `yarn.lock` doesn't align with the `package.json`s (due to missing `yarn.lock` updates in this case) then we can create a `node_modules-` cache entry which will fail integrity checks later.

This explains a few odd scenarios I've seen recently where the e2e jobs all fail because of yarn integrity check failures 🤞